tests/test_transformers.py

```diff
@@ -763,3 +764,118 @@ def test_binarize_works_in_gridsearch(self, train_iris_dataset):
         model = Model(grid)
         result = model.score_estimator(train_iris_dataset)
         assert isinstance(result, Result)
+
+
+class TestRareFeatureEncoder(TransformerBase):
+    def test_rare_feature_encoder_returns_correctly_series(self):
+        threshold = 2
+        fill_rare = "Rare"
+        rare = RareFeatureEncoder(threshold=threshold, fill_rare=fill_rare)
+
+        data = pd.DataFrame({"categorical": [1, "a", "a", 2, "b", 1]})
+        rare.fit(data)
+
+        new_data = pd.DataFrame({"categorical": [1, 1, 1, "a", "b", "b", 3]})
+        result = rare.transform(new_data)
+
+        test_data = pd.DataFrame({"categorical": [1, 1, 1, "a", "Rare", "Rare", 3]})
+
+        assert isinstance(result, pd.DataFrame)
+        assert len(new_data) == len(result)
+        assert test_data.equals(result)
+
+    def test_rare_feature_encoder_returns_correctly_dataframe(self):
+        threshold = 2
+        fill_rare = 99
+        rare = RareFeatureEncoder(threshold=threshold, fill_rare=fill_rare)
+
+        data = pd.DataFrame({"categorical": [1, "a", "a", 2, "b", 2]})
+        rare.fit(data)
+
+        new_data = pd.DataFrame(
+            {
+                "categorical": [1, 1, 1, "a", "b", "b", 3],
+                "numerical": [1, 2, 3, 4, 5, 6, 7],
+            }
+        )
+        result = rare.transform(new_data)
+
+        test_data = pd.DataFrame(
+            {
+                "categorical": [99, 99, 99, "a", 99, 99, 3],
+                "numerical": [1, 2, 3, 4, 5, 6, 7],
+            }
+        )
+
+        assert isinstance(result, pd.DataFrame)
+        assert len(new_data) == len(result)
+        assert test_data.equals(result)
+
+    def test_rare_feature_encoder_handle_nan(self):
```

Comment:
For test code it is fine to have long, descriptive function names - we only see them in the test output and when reading them, so the more descriptive, the better. So maybe something along the lines of:
```suggestion
    def test_rare_feature_encoder_doesnt_count_nans(self):
```
```